src/ngd.ts

```diff
@@ -0,0 +1,197 @@
+// src/handlers/ngd.ts
+
+import { request } from "./utils/ngd/request.js";
+import { get } from "./utils/ngd/request.js";
+import { Feature, type FeatureCollection } from "geojson";
+import { buildUrl } from "./utils/ngd/url.js";
+import { validateParams } from "./utils/validate.js";
+import { BBox, Config } from "./types.js";
+import { initialiseConfig } from "./utils/config.js";
+import fetch from "node-fetch";
+import { getCRS } from "./utils/crs.js";
+
+export { ngd };
+
+// Types
+interface NGDExtent {
+  spatial: {
+    bbox: number[][];
+    crs: string;
+  };
+  temporal: {
+    interval: string[][];
+    trs: string;
+  };
+}
+
+interface NGDLink {
+  href: string;
+  rel: string;
+  type: string;
+  title: string;
+}
+
+interface NGDCollection {
+  id: string;
+  title: string;
+  description: string;
+  crs: string[];
+  storageCrs: string;
+  itemType: string;
+  extent: NGDExtent;
+  links: NGDLink[];
+}
+
+interface NGDCollections {
+  links: NGDLink[];
+  collections: NGDCollection[];
+}
+
+interface NGDQueryables {
+  $schema: string;
+  $id: string;
+  type: string;
+  title: string;
+  description: string;
+  properties: any;
+}
+
+interface NGDSchema {
+  $schema: string;
+  $id: string;
+  type: string;
+  title: string;
+  description: string;
+  extent: NGDExtent;
+  properties: any;
+}
+
+async function requestNGD(config: Config): Promise<FeatureCollection> {
+  return await request(config);
+}
+
+const ngd = {
+  /**
+   * Get NGD features.
+   *
+   * @param {string} apiKey - A valid OS Data Hub key
+   * @param {string} collectionId - A known collection ID
+   * @param {Object} options - Optional arguments
+   * @param {number} [options.offset] - The starting value for the offset
+   * @param {number} [options.limit] - The max number of features to return
+   * @param {string} [options.crs] - CRS of return geoJSON (epsg number or full string e.g. "epsg:27700") [default WGS84]
+   * @param {number[]} [options.bbox] - Bounding box [left, bottom, right, top]
+   * @param {string | number} [options.bboxCRS] - CRS of bounding box (epsg number or full string e.g. "epsg:27700") [default WGS84]
+   * @param {string} [options.filter] - CQL filter string
+   * @param {string} [options.filterCRS] - CRS used if filter contains spatial operation (epsg number or full string e.g. "epsg:27700") [default WGS84]
+   * @param {string} [options.datetime] -  A valid date-time with UTC time zone (Z) or an open or closed interval e.g. 2021-12-12T13:20:50Z
+   * @return {Promise<OSFeatureCollection>} - A GeoJSON Feature Collection
+   */
+  features: async (
+    apiKey: string,
+    collectionId: string,
+    {
+      offset = 0,
+      limit = 1000,
```

Comment:
Should defaults be different between python and js? limit in python is 100.

Reply:
Have set it to 100 on JS - think it's better to be on the low side since it's charged per request